File: frontend/src/lib/ai.ts
import { apiRequest } from "./api";

export type SectionScore = {
  score: number;
  strengths: string[];
  problems: string[];
  recommendations: string[];
};

export type RecommendedRole = {
  role: string;
  match_percentage: number;
  reason: string;
};

export type AIRecommendations = {
  section_scores: Record<string, SectionScore>;
  strengths: string[];
  weaknesses: string[];
  priority_improvements: string[];
  recommended_roles: RecommendedRole[];
};

export const SECTION_SCORE_LABELS: Record<string, string> = {
  summary: "Summary",
  education: "Education",
  skills: "Skills",
  experience: "Experience",
  projects: "Projects",
  certifications: "Certifications",
  achievements: "Achievements",
};

export async function getAIRecommendations(resumeText: string): Promise<AIRecommendations> {
  return apiRequest<AIRecommendations>("/ai/recommendations", {
    method: "POST",
    body: { resume_text: resumeText },
  });
}

File: backend/app/schemas/ai.py
"""Pydantic request/response schemas for AI-generated resume recommendations."""
from pydantic import BaseModel, Field


class AIRecommendationsRequest(BaseModel):
    resume_text: str = Field(min_length=1)


class SectionScore(BaseModel):
    score: int
    strengths: list[str]
    problems: list[str]
    recommendations: list[str]


class RecommendedRole(BaseModel):
    role: str
    match_percentage: int
    reason: str


class AIRecommendationsResponse(BaseModel):
    section_scores: dict[str, SectionScore]
    strengths: list[str]
    weaknesses: list[str]
    priority_improvements: list[str]
    recommended_roles: list[RecommendedRole]
